canonicalwebteam/flask_base/app.py

```diff
@@ -0,0 +1,29 @@
+import flask
+import talisker.flask
+
+from werkzeug.contrib.fixers import ProxyFix
+from werkzeug.debug import DebuggedApplication
+
+from canonicalwebteam.flask_base.converters import RegexConverter
+
+
+class FlaskBase(object):
+    def __init__(self, app=None, service=None):
+        self.app = app
+        if app is not None:
+            self.init_app(app, service)
+
+    def init_app(self, app, service=None):
+        app.config.from_object("canonicalwebteam.flask_base.config.BaseConfig")
+
+        app.url_map.strict_slashes = False
+        app.url_map.converters["regex"] = RegexConverter
+
+        app.wsgi_app = ProxyFix(app.wsgi_app)
+
+        if app.debug:
+            app.wsgi_app = DebuggedApplication(app.wsgi_app)
+
+        talisker.flask.register(app)
+        if service:
```

Comment:
Should `service` ever be allowed to be `None`? Surely every one of our sites should have it set? In which case it should be enforced, no?

Reply:
to be honnest I don't really mind having it as an optionnal parameter if we build smaller projects etc. But I understand that to force us we could set as non optionnal to make sure we track all of our website.

your decision @jpmartinspt 	 :)

Reply:
We can enforce it, I was not sure about this since we currently only monitor bigger websites with greylog but I guess it doesn't hurt to have it set in case we ever need.

Reply:
I don't see the harm. Even if we're just creating a small test project, adding `service="stupidtestapp"` is hardly difficult, or computationally intensive. And presumably if we forget to add it to projects, the risk is we have logs that won't be categorised, that we might only realise we can't find down the line when we need them.